pywhy_stats/kci.py

```diff
@@ -0,0 +1,393 @@
+from functools import partial
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+from numpy._typing import ArrayLike
+from scipy import stats
+from sklearn.metrics.pairwise import rbf_kernel
+
+from pywhy_stats.kernels import delta_kernel, estimate_squared_sigma_rbf
+from pywhy_stats.pvalue_result import PValueResult
+
+
+def ind(
+    X: np.ndarray,
+    Y: np.ndarray,
+    kernel_X: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+    kernel_Y: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+    approx: bool = True,
+    null_sample_size: int = 1000,
+    threshold: float = 1e-5,
+    normalize_data: bool = True,
+) -> PValueResult:
+    """
+    Test whether X and Y are unconditionally independent using the Kernel Independence Tests.
+
+    For testing independence on continuous data, we leverage kernels :footcite:`Zhang2011` that
+    are computationally efficient.
```

Comment:
Also not super important cuz this is a big PR already, but if you look at the `fisherz.py` file it shows how we can layer docstrings in the file, so we don't need to replicate as much across the `ind` and `condind` function. We can add additional docstring at the top of the `kci.py` file before the first release.

Reply:
Not sure if I understand what you mean with the layering. But yea, we can just have follow up to optimize the documentation.

Reply:
Just meant we can add docstrings to top of the file and also inside public-facing functions as well, which will auto-render. We can tackle this when we look at the documentation page before release to see if anything could be better.